lib/semantic_puppet/version.rb

```diff
@@ -30,7 +30,7 @@ def self.valid?(ver)
         false
       else
         prerelease = match[4]
-        prerelease.nil? || prerelease.split('.').all? { |x| !(x =~ /^0\d+/) }
+        prerelease.nil? || prerelease.split('.').all? { |x| (x =~ /^[0-9A-Za-z-]+$/) && !(x =~ /^0\d+$/) }
```

Comment:
Actually, looking closer, the original test was fine and this commit isn't needed. The case with [0-9A-Za-z-] is already covered at line 29. Testing it again here adds nothing.